mobile/src/features/WalletManager/context/WalletManagerProvider.tsx

```diff
@@ -72,6 +72,15 @@ export const WalletManagerProvider: React.FC<
     [actions, walletManager],
   )
 
+  React.useEffect(() => {
```

Comment:
I think moving to the business wrapper and waiting to render anything only after it has hydrated will be safer

Reply:
Moved it to a wrapper that goes before the provider inside the business shell, so that nothing gets rendered until hydration is done or failed.